src/poetry/installation/chef.py

```diff
@@ -81,7 +83,18 @@ def get_cached_archives_for_link(self, link: Link) -> List[Link]:
         links = []
         for archive_type in archive_types:
             for archive in cache_dir.glob(f"*.{archive_type}"):
-                links.append(Link(archive.as_uri()))
+
+                accept = True
+
+                if link.hash:
+                    real_hash = get_file_hash(archive, link.hash_name)
+                    if real_hash != link.hash:
+                        logger = logging.getLogger(__name__)
```

Comment:
loggers are usually declared on top of files as globals. Also, you need to add the logger to `loggers` in `src.poetry.console.application` in line 234.

Reply:
I did, but after adding the logger to `src.poetry.console.application`, the messages began to be duplicated
![image](https://user-images.githubusercontent.com/3264643/169863697-7c8b668f-336d-494f-9faf-2c6eb3f68446.png)